### PR TITLE
Enhance MessageSource to control printing header

### PR DESCRIPTION
**Pull request by aka2en**
This would be useful when users want to customize the output of pevents.

### CODE REVIEW

File: sparta/sparta/pevents/PeventCollector.hpp [hunk 1]
```diff
@@ -180,14 +188,14 @@ namespace pevents{
          * \brief Override the generateCollectionString_() of the bases PairCollector,
          * In this method, we use the pair_cache owned by PairCache.
          */
-        virtual void generateCollectionString_() override final
+        virtual void generateCollectionString_() override
```
Comment (aka2en) [inline sparta/sparta/pevents/PeventCollector.hpp:191]: Removing `final` as users may want to define customized string.

File: sparta/src/Destination.cpp [hunk 2]
```diff
@@ -51,26 +51,29 @@ void DefaultFormatter::write(const sparta::log::Message& msg)
 {
     std::ios::fmtflags f = stream_.flags();
 
-    stream_ << '{';
+    if(msg.print_info)
```
Comment (aka2en) [inline sparta/src/Destination.cpp:54]: Users may want to output headerless message

Comment (knute-sifive) [reply sparta/src/Destination.cpp:54]: This kinda mimics the `sparta::log::RawFormatter`'s behavior...

**Review by knute-sifive — COMMENTED** [review]
I _think_ see what you're doing here, but did you try this with the simulator?
```
--pevents pevents.log.raw ALL
```

**Comment by aka2en** [comment]
`sparta::log::RawFormatter` can achieve what I want!

**Pull request closed without merging**